libraries/dagster-iceberg/src/dagster_iceberg/io_manager/spark.py

```diff
@@ -83,11 +83,14 @@ def connect(
         table_slice: TableSlice,
     ) -> Iterator[SparkSession]:
         builder = cast(SparkSession.Builder, SparkSession.builder)
-        if (
-            context.resource_config is not None
-            and (spark_config := context.resource_config["spark_config"]) is not None
-        ):
-            builder.config(map=cast(dict[str, "OptionalPrimitiveType"], spark_config))
+        if context.resource_config is not None:
+            if (spark_config := context.resource_config["spark_config"]) is not None:
+                builder.config(
+                    map=cast(dict[str, "OptionalPrimitiveType"], spark_config)
+                )
+
+            if (remote_url := context.resource_config["remote_url"]) is not None:
+                builder.remote(cast(str, remote_url))
```

Comment:
this is explicitly overriding the behavior from the config right? But the user can technically still set it?

I'd leave some sort of comment to indicate what the preferred precedence order is here so that we don't accidentally break it. Maybe even error on collision (but i think that just being explicit about the inheritance order is fine)

Reply:
Correct. I would expect it to be unlikely that people set `spark.remote` via `config()` (that's not how it's usually documented by Spark), but it's technically possible and was using it as a workaround before. I can try to document the precedence (including `SPARK_REMOTE` env var); I'll create a ticket because I think this is less important right now, as it largely just depends on how Spark handles it.